src/debugger/appWorker.ts

```diff
@@ -203,6 +206,9 @@ export class MultipleLifetimesAppWorker {
             if (object.method === "prepareJSRuntime") {
                 // The MultipleLifetimesAppWorker will handle prepareJSRuntime aka create new lifetime
                 this.gotPrepareJSRuntime(object);
+            } else if (object.method === "$disconnected") {
+                // We need to shutdown the current app worker, and create a new lifetime
+                this.singleLifetimeWorker = null;
```

Comment:
Node doesn't give us any way to "stop" a sandbox, because it's not a real sandbox. It's more like using a different namespace for the global variables only. I don't think we have any easy way of stopping a timer or something similar, if it was started inside the RN app code.

Thankfully, react-native seems to be overriding methods like setTimeout, etc.. https://github.com/facebook/react-native/blob/master/Libraries/JavaScriptAppEngine/System/JSTimers/JSTimers.js and they seem to be doing some sort of cleanup. I tested putting this in my app, and it worked perfectly with live reloading: 
        setInterval(function () {
            console.log('test: ' + new Date());
        }, 5000);

So I'm guessing that at least for timers, RN is taking care of that.

Regarding the websocket, RN seems to support several "lifetimes" in the same socket, so we don't need to close it.